zebra-network/src/meta_addr.rs

```diff
@@ -274,7 +273,7 @@ impl MetaAddr {
     pub fn new_gossiped_change(self) -> MetaAddrChange {
         NewGossiped {
             addr: canonical_socket_addr(self.addr),
-            untrusted_services: self.services,
+            untrusted_services: self.services.expect("unexpected missing services"),
```

Comment:
Test Coverage: Do we have tests that make sure that this expectation is always satisfied?

Since `services` is `pub(crate)`, it will be difficult to test all the code that could possibly modify the field. So if we're not certain about our test coverage, let's use the default `NODE_NETWORK` services here.

This is a merge blocker, because it could cause test or runtime panics.

```suggestion
            untrusted_services: self.services.unwrap_or(PeerServices::NODE_NETWORK),
```

Reply:
I tried to solve this with the help of the compiler. I made the `new_gossiped_change()` method return `Option<MetaAddrChange>` so that it can return `None` if the field is missing. The caller can then handle the case.

For now the only case is `CandidateSet::send_addrs` (BTW that seems to be a weird name for the method :thinking:) which adds the addresses gossiped from another peer. I changed that method to ignore peers without `services`, but that should be impossible. Not sure if it would make sense to make it `panic` if that happens there?

Reply:
If we're sure that something can never happen, and we have tests for it, we should `expect` or panic if it happens.

So I think we should panic if `CandidateSet::send_addrs` gets `None` services. Because all of those `MetaAddr`s should have been parsed from `addr` or `addrv2` messages, which always have a services field.

Reply:
Done.